test/phoenix/router/forward_test.exs

```diff
@@ -0,0 +1,144 @@
+defmodule Phoenix.Router.ForwardTest do
+  use ExUnit.Case, async: true
+  use RouterHelper
+
+  defmodule Controller do
```

Comment:
The issue is the forward process does the script_name dance, so if we call the helpers in the test before/after the request, the conn is not in the context of the forwarded plug?